lib/data/multipleBackendGateway.js

```diff
@@ -42,49 +111,106 @@ const multipleBackendGateway = {
                 const dataRetrievalInfo = {
                     key,
                     dataStoreName: controllingLocationConstraint,
+                    dataStoreType: client.clientType,
                 };
                 return callback(null, dataRetrievalInfo);
             });
     },
 
     get: (objectGetInfo, range, reqUids, callback) => {
-        const client = clients[objectGetInfo.dataStoreName];
+        let key;
+        let client;
+        // for backwards compatibility
+        if (typeof(objectGetInfo) === 'string') {
+            key = objectGetInfo;
+            client = clients.legacy;
+        } else {
+            key = objectGetInfo.key;
+            client = clients[objectGetInfo.dataStoreName];
+        }
         if (client.clientType === 'scality') {
-            return client.get(objectGetInfo.key, range, reqUids, callback);
+            return client.get(key, range, reqUids, callback);
+        }
+        if (client.clientType === 'aws_s3') {
+            const stream = client.getObject({
+                Bucket: client.awsBucketName,
+                Key: key,
+                Range: range,
+            }).createReadStream().on('error', err => {
+                const log = createLogger(reqUids);
+                log.error('error streaming data', { error: err,
+                    dataStoreName: client.dataStoreName });
+                return callback(errors.InternalError);
+            });
+            return callback(null, stream);
         }
         return client.get(objectGetInfo, range, reqUids, callback);
     },
 
     delete: (objectGetInfo, reqUids, callback) => {
-        const client = clients[objectGetInfo.dataStoreName];
+        let key;
+        let client;
+        // for backwards compatibility
+        if (typeof(objectGetInfo) === 'string') {
+            key = objectGetInfo;
+            client = clients.legacy;
+        } else {
+            key = objectGetInfo.key;
+            client = clients[objectGetInfo.dataStoreName];
+        }
+        if (client.clientType === 'aws_s3') {
+            const params = {
+                Bucket: client.awsBucketName,
+                Key: key,
+            };
+            return client.deleteObject(params, err => {
+                if (err) {
+                    const log = createLogger(reqUids);
+                    log.error('error deleting object from datastore',
+                    { error: err, implName: client.clientType });
+                    return callback(errors.InternalError);
+                }
+                return callback();
+            });
+        }
         if (client.clientType === 'scality') {
-            return client.delete(objectGetInfo.key, reqUids, callback);
+            return client.delete(key, reqUids, callback);
         }
         return client.delete(objectGetInfo, reqUids, callback);
     },
 
     healthcheck: (log, callback) => {
```

Comment:
Seeing this function getting more complex, maybe time to think making `healthcheck` another client backend-specific call?